include/AST/LetDecl.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef GLU_AST_DECL_LETNDECL_HPP
+#define GLU_AST_DECL_LETNDECL_HPP
+
+#include "ASTNode.hpp"
+#include "Types/Types.hpp"
+
+#include <string>
+
+namespace glu::ast {
+
+/// @class LetDecl
+/// @brief Represents a let declaration in the AST.
+///
+/// This class inherits from DeclBase and encapsulates the details of a 'let'
+/// declaration, including its name, type, and value.
+class LetDecl : public DeclBase {
+    std::string _name;
+    glu::types::TypeBase *_type;
+    glu::ast::ASTNode *_value;
+
+public:
+    /// @brief Constructor for the LetDecl class.
+    /// @param location The source location of the declaration.
+    /// @param parent The parent AST node.
+    /// @param name The name of the declared variable.
+    /// @param type The type of the declared variable.
+    /// @param value The value assigned to the declared variable.
+    LetDecl(
+        SourceLocation location, ASTNode *parent, std::string name,
+        glu::types::TypeBase *type, ASTNode *value
+    )
```

Comment:
you can't put both parent and children in the constructor... we either create parents then children, or children then parents. Creating the children and setting the parent in the constructor is probably easier, so never set the parent in constructors?

Reply:
I agree that the parent of `value` is `this`, but here I need the parent because the class inherits from `DeclBase`, which itself needs a parent, so I don't really have a choice but to use one in the constructor